lib/AST/GenericSignature.cpp

```diff
@@ -603,7 +604,14 @@ SmallVector<Requirement, 4> GenericSignature::requirementsNotSatisfiedBy(
 
   // Find the requirements that aren't satisfied.
   for (const auto &req : getRequirements()) {
-    if (!otherSig->isRequirementSatisfied(req))
+    auto reqToCheck = req;
+    if (subMap) {
+      auto subsReq = req.subst(subMap);
+      if (subsReq.hasValue()) {
```

Comment:
If the substitution failed... I'm not sure what to do. Using the unsubstituted requirement is clearly wrong. Maybe just return assert in this case, and we'll rethink it if/when we hit it?

Reply:
Sure, I have added an assert!